pkg/planner/core/casetest/testdata/integration_suite_out.json

```diff
@@ -1245,13 +1245,14 @@
           "└─TableReader 1.00 root  MppVersion: 2, data:ExchangeSender",
           "  └─ExchangeSender 1.00 mpp[tiflash]  ExchangeType: PassThrough",
           "    └─HashAgg 1.00 mpp[tiflash]  funcs:count(1)->Column#9",
-          "      └─HashJoin 80000000.00 mpp[tiflash]  CARTESIAN inner join",
-          "        ├─ExchangeReceiver(Build) 8000.00 mpp[tiflash]  ",
-          "        │ └─ExchangeSender 8000.00 mpp[tiflash]  ExchangeType: Broadcast, Compression: FAST",
-          "        │   └─Projection 8000.00 mpp[tiflash]  1->Column#8",
-          "        │     └─Selection 8000.00 mpp[tiflash]  gt(mul(test.t1.c1, test.t1.c1), plus(test.t1.c2, 10))",
-          "        │       └─TableFullScan 10000.00 mpp[tiflash] table:ta pushed down filter:empty, keep order:false, stats:pseudo",
-          "        └─TableFullScan(Probe) 10000.00 mpp[tiflash] table:tb keep order:false, stats:pseudo"
+          "      └─Projection 80000000.00 mpp[tiflash]  Column#8",
```

Comment:
Why is there an extra projection here?

Reply:
The join of the tiflash side doesn't support inline projection in itself. So we need to add the extra proj when the inline projection happens

Reply:
And before this pr, tiflash side doesn't prune any column for the join. It's actually optimizing the join's output.